polkadot/node/subsystem-bench/src/lib/network.rs

```diff
@@ -349,8 +349,20 @@ impl NetworkInterface {
 					task_tx_limiter.lock().await.reap(size).await;
 
 					match peer_message {
-						NetworkMessage::MessageFromNode(peer, message) =>
-							tx_network.send_message_to_peer(&peer, message),
+						NetworkMessage::MessageFromNode(peer, message) => {
+							let tx_network = tx_network.clone();
+							let send_task = async move {
+								let receiver = tx_network.peer(&peer);
+								let latency_ms = receiver.latency_ms();
+								if latency_ms > 0 && receiver.is_connected() {
+									emulate_latency(latency_ms).await;
+								}
+								tx_network.send_message_to_peer(&peer, message);
```

Comment:
Why do we still want to send to a disconnected peer ? Wouldn't the assert in `send_message_to_peer` trigger if peer is not connected ?